packages/java/endpoint/src/main/java/com/vaadin/hilla/crud/filter/FilterTransformer.java

```diff
@@ -0,0 +1,118 @@
+package com.vaadin.hilla.crud.filter;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+
+/**
+ * Remaps property names in filters and pageable objects.
+ */
+public class FilterTransformer {
+
+    private final Map<String, String> mappings = new HashMap<>();
+    private Function<PropertyStringFilter, PropertyStringFilter> filterTransformation;
+
+    /**
+     * Declares a mapping from one property name to another. If a filter or
+     * pageable is remapped, all occurrences of the property name will be
+     * replaced with the new name.
+     *
+     * @param from
+     *            The original property name.
+     * @param to
+     *            The new property name.
+     * @return This instance.
+     */
+    public FilterTransformer withMapping(String from, String to) {
+        mappings.put(from, to);
+        return this;
+    }
+
+    /**
+     * Declares a function that will be applied to all
+     * {@link PropertyStringFilter} instances. This can be used to modify the
```

Comment:
This description of `function that will be applied to all {@link PropertyStringFilter} instances.` is a bit vague to me. If it means that this function takes the filter instance as a parameter and can effect all the filter properties and and their values, could we clarify the javadoc to reflect this?